### PR TITLE
Add nots stats service

### PR DESCRIPTION
**Pull request by CalumMcCall**
## What's new in this PR?

This PR adds the service used to store statistics about fcm notifications.

### CODE REVIEW

File: zmessaging/src/main/scala/com/waz/service/push/FCMNotificationStatsService.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+class FCMNotificationStatsServiceImpl(fcmTimestamps: FCMNotificationsRepository,
+                                      fcmStats: FCMNotificationStatsRepository)
+  extends FCMNotificationStatsService {
+
+  import FCMNotificationStatsService._
+
+  private implicit val ec: ExecutionContext = Threading.Background
+
+  override def storeNotificationState(id: Uid, stage: String, timestamp: Instant): Future[Unit] =
```
Comment (yeryomenkom) [inline zmessaging/src/main/scala/com/waz/service/push/FCMNotificationStatsService.scala]: Can we rewrite this method without nested flatmaps? With `for`.

File: zmessaging/src/main/scala/com/waz/service/push/FCMNotificationsRepository.scala [hunk 2]
```diff
@@ -17,48 +17,78 @@
...
+
+object FCMNotificationsRepository {
 
   val Pushed = "pushed"
   val Fetched = "fetched"
   val StartedPipeline = "startedPipeline"
+  val FinishedPipeline = "finishedPipeline"
+  val everyStage: Seq[String] = Seq(Pushed, Fetched, StartedPipeline, FinishedPipeline)
+
+  def prevStage(stage: String): Option[String] = stage match {
```
Comment (yeryomenkom) [inline zmessaging/src/main/scala/com/waz/service/push/FCMNotificationsRepository.scala]: I think that such method should live near the model definition. Not in repository.

**Comment by CalumMcCall** [comment]
build again

**Comment by CalumMcCall** [comment]
build again

File: zmessaging/src/main/scala/com/waz/service/push/PushService.scala [hunk 3]
```diff
@@ -183,9 +184,12 @@ class PushServiceImpl(selfUserId:           UserId,
   wsPushService.connected().onChanged.map(WebSocketChange).on(dispatcher)(syncHistory(_))
 
   private def storeNotifications(notifications: Seq[PushNotificationEncoded]): Future[Unit] =
-    Serialized.future(PipelineKey)(notificationStorage.saveAll(notifications).flatMap { _ =>
-      notifications.lift(notifications.lastIndexWhere(!_.transient)).fold(Future.successful(()))(n => idPref := Some(n.id))
-    })
+    Serialized.future(PipelineKey)(
```
Comment (yeryomenkom) [inline zmessaging/src/main/scala/com/waz/service/push/PushService.scala]: Can be written clearer with async/await.

**Pull request merged**